### DISCUSSION TITLE
Should MutableCompoundShape::AdjustCenterOfMass also recalculate subshape bounding boxes?

### DISCUSSION THREAD

**Discussion by bjornbytes** [jrouwe/JoltPhysics#1168] [General]
Hi!  I'm running into an issue where adjusting the center of mass of a `MutableCompoundShape` leads to incorrect collision detection.  It seems like `AdjustCenterOfMass` moves the subshapes around the new COM but doesn't update their bounding boxes.

Here's a small program that demonstrates the problem:

```c++
#include <Jolt/Jolt.h>
#include <Jolt/Physics/Collision/Shape/BoxShape.h>
#include <Jolt/Physics/Collision/Shape/MutableCompoundShape.h>
#include <Jolt/Physics/Collision/CollidePointResult.h>
#include <Jolt/Physics/Collision/CollisionCollectorImpl.h>
#include <iostream>

using namespace JPH;

int main(int argc, char** argv)
{
  RegisterDefaultAllocator();

  MutableCompoundShapeSettings settings;
  settings.AddShape(Vec3(-1.f, 0.f, 0.f), Quat::sIdentity(), new BoxShape(Vec3(1.f, 1.f, 1.f)));
  MutableCompoundShape* shape = static_cast<MutableCompoundShape*>(settings.Create().Get().GetPtr());

  shape->AddShape(Vec3(1.f, 0.f, 0.f), Quat::sIdentity(), new BoxShape(Vec3(1.f, 1.f, 1.f)));
  shape->AdjustCenterOfMass();

  // Comment this in to update the AABBs and get 1 hit instead of 0
  //shape->ModifyShape(1, Vec3(1.f, 0.f, 0.f), Quat::sIdentity());

  AllHitCollisionCollector<CollidePointCollector> collector;
  shape->CollidePoint(Vec3(0.5f, 0.f, 0.f), SubShapeIDCreator(), collector);
  std::cout << collector.mHits.size() << std::endl;

  return 0;
}
```

After `AdjustCenterOfMass`, Jolt still thinks the bounding box of the second cube is centered around `(2, 0, 0)` instead of `(1, 0, 0)`.

Since AFAIK there's no way to manually update the subshape bounding boxes, the only workarounds I can think of are kind of hacky: add and remove a dummy shape, move an existing shape to the position it's already at, etc.  It seems like adjusting the center of mass should take care of updating the subshape bounds since it moves the shapes.

**Comment by jrouwe** [comment]
Thanks for reporting this! This is a bug and I've created a fix in #1169 (currently going through the build).

**Reply by jrouwe** [reply]
Note that `CollidePoint` takes a point relative to the COM, see: https://jrouwe.github.io/JoltPhysics/#center-of-mass. You may want to do:

```
shape->CollidePoint(Vec3(0.5f, 0.f, 0.f) - shape->GetCenterOfMass(), SubShapeIDCreator(), collector);
```

**Reply by jrouwe** [reply]
The fix has been merged.

**Reply by bjornbytes** [reply]
Thanks for the quick fix!
